client/clientservice/src/state_snapshot_service.cpp

```diff
@@ -112,6 +112,10 @@ static void getResponseSetStatus(concord::client::concordclient::SendResult&& se
         LOG_INFO(logger, "Request failed with OVERLOADED error for cid=" << correlation_id);
         return_status = grpc::Status(grpc::StatusCode::UNAVAILABLE, "All clients occupied");
         break;
+      case (static_cast<uint32_t>(bftEngine::OperationResult::EXEC_ENGINE_REJECT_ERROR)):
+        LOG_INFO(logger, "Request failed with EXEC_ENGINE_REJECT_ERROR error for cid=" << correlation_id);
+        return_status = grpc::Status(grpc::StatusCode::UNAVAILABLE, "Aborted");
```

Comment:
Why do we return UNAVAILABLE here? @arc-vmware Does this make sense to you?
The [documentation](https://github.com/vmware/concord-bft/blob/master/client/proto/state_snapshot/v1/state_snapshot.proto) says that Concord is not ready to process a request in this case.

Reply:
I think making it UNAVAILABLE doesn't make sense, since CS is available here. So we should change this to:
return_status = grpc::Status(grpc::StatusCode::UNKNOWN, "Execution rejected this request");
For the UNAVAILABLE case, the caller might retry after some time with the same req. But in this case, the request itself is a problem, so instead of retry, they should crash. So I agree with @robem that this should be changed.

Reply:
@robem - Sure. I can change the error. But for CONFLICT_DETECTED also we return UNAVAILABLE here. So wanted to confirm if that also needs to be UNKNOWN?

Reply:
We should be consistent and return `ABORTED` as well, IMHO. If the same error on two different services returns two different errors then this is somewhat confusing.

Reply:
@cjn why EXEC_ENGINE_REJECT_ERROR and CONFLICT_DETECTED are related?

Reply:
We can return UNKNOWN and UNAVAILABLE. 
UNAVAILABLE should be returned in cases where retrying the same message can succeed. UNKNOWN means the client of CS should crash and seek help from SRE.

Reply:
As discussed changed to UNKNOWN